Bugfix: Remove illegal chars

joeflack4 opened the pull request:
## Updates
Remove illegal unicode characters found.

## Related
- https://github.com/monarch-initiative/mondo-ingest/pull/412
- https://github.com/monarch-initiative/mondo-ingest/pull/434

---

@twhetzel Thanks for helping narrow the problem!

joeflack4 committed:
Bugfix: Remove illegal chars

joeflack4 commented on icd11_foundation_ingest/clean_illegal_chars.py:
```diff
@@ -0,0 +1,23 @@
+    """Clean"""
+    with open(inpath, 'rb') as file:
+        content = file.read()
+    cleaned_content = content.replace(b'\x03', b'')
```
The fix

joeflack4 commented on makefile:
```diff
@@ -0,0 +1,36 @@
+	wget ${SOURCE_URL} -O $@
+
+# Requires GitHub CLI: https://cli.github.com/
+deploy-release: | tmp/output/release/
```
Deploy release command using GitHub CLI

joeflack4 replied:
Release is up: https://github.com/monarch-initiative/icd11/releases
Updating `mondo-ingest` now to use.

joeflack4 commented on makefile:
```diff
@@ -0,0 +1,36 @@
+all: tmp/output/release/icd11foundation.owl
+
+tmp/output/release/icd11foundation.owl: tmp/input/source.owl | tmp/output/release/
+	 python3 icd11_foundation_ingest/clean_illegal_chars.py -i $< -o $@
```
Goal for fix and release artefact. Also runs when running `make all`.

joeflack4 replied:
[Trish wrote](https://github.com/monarch-initiative/icd11/pull/2#issuecomment-1949112751):
> @joeflack4 while I haven't tried to run this yet, looks like this solves the issues. While I'm not opposed to the python script, did you also have a chance to try adding a unix call, e.g. `tr '\003' ' ' < input_file > output_file`, into the workflow which is a bit more straightforward

That is a good idea; I shoulda reached for a unix command, I've never actually seen `tr` before. Anyway as we spoke about on the call, there are more such illegal chars, so my first priority is to reach for a unix solution that removes all such illegal unicode chars.

twhetzel replied:
Here's a more general Unix command (from ChatGPT) that works by deleting all characters that are not in the ASCII range. When `-cd` are used together, `-c` complements the set, and `-d` deletes the characters that match the resulting set. This combination effectively removes all characters from the input stream that are not in the specified ASCII range, effectively removing Unicode characters.

`tr -cd '\11\12\15\40-\176' < yourfile.txt > yourfile-without-unicode.txt`

I ran this on "component-download-icd11foundation.owl.owl" and the resulting file could be opened by Protege and in a quick look over the labels they seem fine.

joeflack4 replied:
@twhetzel That's amazing! ChatGPT is a good old chap. Thanks for the help. I'll use this solution!

joeflack4 replied:
Worked like a charm! `mondo-ingest` has no trouble parsing now.

This can be merged unless there are other requests for tweaks.

twhetzel commented:
@joeflack4 while I haven't tried to run this yet, looks like this solves the issues. While I'm not opposed to the python script, did you also have a chance to try adding a unix call, e.g. `tr '\003' ' ' < input_file > output_file`, into the workflow which is a bit more straightforward

joeflack4 committed:
Bugfix: Remove all illegal chars
Previous commit involved Python based approach which just assumed 1 unicode char was present. This was not the case. New approach uses unix command and removes all such characters.

joeflack4 commented:
This has been ready to merge, and I'm merging it now for ease of making this new PR (#3) to address [equivalence issues](https://github.com/monarch-initiative/mondo-ingest/pull/412#discussion_r1493469285).

twhetzel commented:
This looks ok.
